src/iam.ts

```diff
@@ -98,6 +98,21 @@ export class IAM extends IAMBase {
 
     return { isMetamaskPresent: !!provider, chainId };
   }
+
+  static isSessionActive() {
+    if (window && window.sessionStorage) {
+      return Boolean(sessionStorage.getItem(PUBLIC_KEY));
```

Comment:
Just curious, is there is a reason in particular why having the publicKey set is more indicative of the session state than the provider? I think this definitely looks good and works as is, just a trying to understand a little better

Reply:
Yeah I think you're right to be able to return true here we need to have both information like publicKey and provider type, I'll change that 👍